guides.Rmd

```diff
@@ -196,136 +317,260 @@ Note the use of `%o%` to quickly generate the multiplication table, and that the
+
+*   `axis.colour`, `axis.linewidth` and `axis.arrow` are used to
+    control the guide axis that is displayed alongside the legend keys 
+
+    ```{r}
+    base + guides(size = guide_bins(show.limits = TRUE))
+    base + guides(
+      size = guide_bins(
+        axis.colour = "red",
+        axis.arrow = arrow(
+          length = unit(.1, "inches"), 
+          ends = "first", 
+          type = "closed"
+        )
+      )
+    )
+    ```
+
+*   `keywidth`, `keyheight`, `reverse` and `override.aes` have the same
+  behaviour for `guide_bins()` 
+
+### `guide_colourbar`
+
+The colour bar guide is designed for continuous ranges of colors---as its name implies, it outputs a rectangle over which the color gradient varies.  The most important arguments are: \index{Legend!colour bar} \index{Colour bar}
+
+* `barwidth` and `barheight` (along with `default.unit`) allow you to specify
+   the size of the bar. These are grid units, e.g. `unit(1, "cm")`.
+  
+* `nbin` controls the number of slices. You may want to increase this from
+  the default value of 20 if you draw a very long bar.
+
+* `reverse` flips the colour bar to put the lowest values at the top.
+
+These options are illustrated below:
+
+`r columns(3)`
+```{r}
+base <- ggplot(mpg, aes(cyl, displ, colour = hwy)) +
+  geom_point(size = 2)
+
+base
+base + guides(colour = guide_colourbar(reverse = TRUE))
+base + guides(colour = guide_colourbar(barheight = unit(1, "cm")))
 ```
 
+### `guide_coloursteps()`
+
+This "colour steps" guide is a version of `guide_colourbar()` appropriate for binned colour and fill scales. It shows the area between breaks as a single constant colour, rather than displaying a colour gradient that varies smoothly along the bar. Arguments mostly mirror those for `guide_colourbar()`. The additional arguments are as follows:
+
+*   `show.limits` indicates whether values should be shown at the ends of the stepped colour bar (analogous to the corresponding argument in `guide_bins()`)
+
+    `r columns(2)`
+    ```{r}
+    base <- ggplot(mpg, aes(displ, hwy, colour = cyl)) + 
+      geom_point() + 
+      scale_color_binned()
+
+    base + guides(colour = guide_coloursteps(show.limits = TRUE))
+    base + guides(colour = guide_coloursteps(show.limits = FALSE))
+    ```
+
+*   `ticks` is a logical variable indicating whether tick marks should be displayed adjacent to the legend labels (default is `NULL`, in which case the value is inherited from the scale)
 
+*   `even.steps` is a logical variable indicating whether bins should be evenly spaced (default is `TRUE`) or proportional in size to their frequency in the data
 
-## Legends for multiple layers {#sub-layers-legends}
+
+
+### `guide_axis()`
+
+Guide functions exist mostly to control plot legends, but---as legends and axes are both kinds of guide---ggplot2 also supplies a `guide_axis()` function for axes. Its main purpose is to modify thematic properties:
```

Comment:
```suggestion
Guide functions exist mostly to control plot legends, but---as legends and axes are both kinds of guide---ggplot2 also supplies a `guide_axis()` function for axes. Its main purpose is to provide additional controls to prevent labels from overlapping:
```